plugins/defaultplugins/ifplugin/stn_config.go

```diff
@@ -0,0 +1,191 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ifplugin
+
+import (
+	"fmt"
+
+	govppapi "git.fd.io/govpp.git/api"
+	"github.com/ligato/cn-infra/logging"
+	"github.com/ligato/cn-infra/logging/measure"
+	"github.com/ligato/cn-infra/utils/addrs"
+	"github.com/ligato/cn-infra/utils/safeclose"
+	"github.com/ligato/vpp-agent/idxvpp"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/bin_api/stn"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/ifaceidx"
+	model_stn "github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/model/stn"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin/vppcalls"
+	"github.com/ligato/vpp-agent/plugins/govppmux"
+	"github.com/prometheus/common/route"
+)
+
+type StnConfigurator struct {
+	Log logging.Logger
+
+	GoVppmux    govppmux.API
+	StnIndexes  idxvpp.NameToIdxRW
+	StnIndexSeq uint32
+	SwIfIndexes ifaceidx.SwIfIndex
+	vppChan     *govppapi.Channel
+
+	Stopwatch *measure.Stopwatch
+}
+
+// Init initializes ARP configurator
+func (plugin *StnConfigurator) Init() (err error) {
+	plugin.Log.Debug("Initializing StnConfigurator")
+
+	// Init VPP API channel
+	plugin.vppChan, err = plugin.GoVppmux.NewAPIChannel()
+	if err != nil {
+		return err
+	}
+
+	err = plugin.checkMsgCompatibility()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Check STN rule raw data
+func CheckStn(stnInput *model_stn.StnRule, index ifaceidx.SwIfIndex, log logging.Logger) (*vppcalls.StnRule, error) {
+	if stnInput == nil {
+		log.Infof("STN input is empty")
+		return nil, nil
+	}
+	if stnInput.Interface == "" {
+		log.Infof("STN input does not contain interface")
+		return nil, nil
+	}
+	if stnInput.IpAddress == "" {
+		log.Infof("STN input does not contain IP")
+		return nil, nil
+	}
+
+	ifName := stnInput.Interface
+	ifIndex, _, exists := index.LookupIdx(ifName)
+	if !exists {
+		return nil, fmt.Errorf("STN entry interface %v not found", ifName)
+	}
+
+	parsedIP, _, err := addrs.ParseIPWithPrefix(stnInput.IpAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	stn_rule := &vppcalls.StnRule{
+		IpAddress: *parsedIP,
+		IfaceIdx:  ifIndex,
+	}
+	return stn_rule, nil
+}
+
+func (plugin *StnConfigurator) Add(rule *model_stn.StnRule) error {
+	plugin.Log.Infof("Creating new STN rule %v", rule)
+
+	// Check stn data
+	stnRule, err := CheckStn(rule, plugin.SwIfIndexes, plugin.Log)
+	if err != nil {
+		return err
+	}
+	plugin.Log.Debugf("adding STN rule: %+v", stnRule)
+	// Create and register new stn
+	errVppCall := vppcalls.AddStnRule(stnRule.IfaceIdx, &stnRule.IpAddress, plugin.Log, plugin.vppChan, measure.GetTimeLog(stn.StnAddDelRule{}, plugin.Stopwatch))
```

Comment:
stnRule.anything could throw nil pointer in current implementation if CheckStn returns nil, nil